libsoundio fails to build on macOS 12.5.1 (21G83) Mac mini (M1, 2020)

### Steps To Reproduce
Steps to reproduce the behaviour:
1. `nix-shell -p libsoundio` 

#### Expected behaviour

Shell opens with libsoundio available.

#### Actual behaviour

Build fails with the error `In file included from ../src/pulsecore/core-util.c:114:
/nix/store/b9bh3z78rqff14bf0fd78xb07b5cnwdw-clang-wrapper-11.1.0/resource-root/include/cpuid.h:11:2: error: this header is for x86 only`.

Note that even tho error itself belongs to a dependency `libpulseaudio`, I'm reporting `libsoundio` because I assume `libpulseaudio` shouldn't be a dependency when building for macOS, or at least there should be a way to disable it.

### Build log
https://gist.github.com/ul/4df57c6b3d37fdf0704510d88abd8a61

### Additional context
Add any other context about the problem here.

### Notify maintainers
@andrewrk 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.5.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.10.3`
 - channels(ul): `""`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/development/libraries/libsoundio/default.nix
+++ b/pkgs/development/libraries/libsoundio/default.nix
@@ -13,9 +13,13 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
-  buildInputs = [ libjack2 libpulseaudio ]
-    ++ lib.optional stdenv.isLinux alsa-lib
+  buildInputs = [ libjack2 ]
+    ++ lib.optionals stdenv.isLinux [ libpulseaudio alsa-lib ]
     ++ lib.optional stdenv.isDarwin AudioUnit;
+
+  cmakeFlags = lib.optionals stdenv.isDarwin [
+    "-DBUILD_TESTS=OFF"
+  ];
 
   NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin "-Wno-strict-prototypes";
 

